packages/component-aws-s3/src/s3_utils.ts

```diff
@@ -0,0 +1,334 @@
+import { S3Client, GetObjectCommand, PutObjectCommand, DeleteObjectCommand } from '@aws-sdk/client-s3';
+import { getSignedUrl } from '@aws-sdk/s3-request-presigner';
+import { createPresignedPost, PresignedPost } from '@aws-sdk/s3-presigned-post';
+
+import {
+    ArgOptional,
+    Communicator, CommunicatorContext,
+    Configurable,
+    InitContext,
+    Workflow,
+    WorkflowContext
+} from '@dbos-inc/dbos-sdk';
+import { AWSServiceConfig, getAWSConfigForService, getAWSConfigs } from '@dbos-inc/aws-config';
+import { DBOSError } from '@dbos-inc/dbos-sdk/dist/src/error';
+
+export interface FileRecord {
+    key: string;
+}
+
+export interface S3Config{
+    awscfgname?: string,
+    awscfg?: AWSServiceConfig,
+    bucket: string,
+    s3Callbacks: {
+        newActiveFile: (ctx: WorkflowContext, rec: FileRecord) => Promise<unknown>;
+        newPendingFile: (ctx: WorkflowContext, rec: FileRecord) => Promise<unknown>;
+        fileActivated: (ctx: WorkflowContext, rec: FileRecord) => Promise<unknown>;
+        fileDeleted: (ctx: WorkflowContext, rec: FileRecord) => Promise<unknown>;
+    }
+}
+
+@Configurable()
+export class S3Ops {
+    //////////
+    // S3 Configuration
+    //////////
+
+    static AWS_S3_CONFIGURATIONS = 'aws_s3_configurations';
+
+    static async initConfiguration(ctx: InitContext, config: S3Config) {
+        // Get the config and call the validation
+        getAWSConfigs(ctx, S3Ops.AWS_S3_CONFIGURATIONS);
```

Comment:
I had a similar question about not checking the return value of getAWSConfigs in the SES communicator